src/utils/utils_common.h

```diff
@@ -36,45 +35,36 @@ extern "C" {
 
 #define __TLS __declspec(thread)
 
-static inline char *util_getenv(const char *name) {
-    char *buffer;
-    size_t numberOfElements;
-    errno_t err = _dupenv_s(&buffer, &numberOfElements, name);
-    if (err) {
-        return NULL;
-    }
-
-    return buffer;
-}
-
-static inline void util_free_getenv(char *val) { free(val); }
-
 // TODO: implement util_get_page_size() for Windows
 static inline size_t util_get_page_size(void) { return 4096; }
 
 #else /* Linux */
 
 #define __TLS __thread
 
-static inline char *util_getenv(const char *name) { return getenv(name); }
-static inline void util_free_getenv(const char *val) {
-    (void)val; // unused
-}
-
 static inline size_t util_get_page_size(void) { return sysconf(_SC_PAGE_SIZE); }
 
 #endif /* _WIN32 */
 
+// Return value
+// If the function succeeds, the return value is the number of characters
+// stored in the buffer pointed to by buffer, not including
+// the terminating null character.
+//
+// If buffer is not large enough to hold the data, the return value
+// is (-1) * the buffer size (with the minus sign), in characters,
+// required to hold the string and its terminating null character
+// and the contents of buffer are undefined.
+//
+// If the function fails, the return value is zero.
+int util_env_var(const char *envvar, char *buffer, size_t buffer_size);
+
+// Check if the environment variable contains the given string.
+int util_env_var_has_str(const char *envvar, const char *str);
+
 // check if we are running in the proxy library
 static inline int is_running_in_proxy_lib(void) {
```

Comment:
maybe I'm crazy, but... should we have a test for these functions? :D

Perhaps just for the `is_running_in_proxy_lib` function...?

Reply:
We can, but I would wait with it, because there already is a PR removing this function at all: https://github.com/oneapi-src/unified-memory-framework/pull/263